workflow_support_scripts/convert_curve_files.py

```diff
@@ -0,0 +1,92 @@
+
+    # Check the input folder
+    if not os.path.exists(input_folder):
+        exit(f"Input folder {input_folder} does not exist")
+
+    # Set up the output folder
+    if os.path.exists(output_folder):
+        if input("This will overwrite the output folder, continue? Y/N ").lower()[0] == "y":
+            for root, _, files in os.walk(output_folder):
+                for output_file in files:
+                    os.remove(os.path.join(root, output_file))
+        else:
+            exit(
+                f"Output folder {output_folder} has not been"
+                f" overwritten and calibration files have not been converted."
+            )
+    else:
+        print(f"Create directory {output_folder}")
+        os.mkdir(output_folder)
+
+    # Go through all folders in input_folder to find curve file
+    for root, _, files in os.walk(input_folder):
+        # Find curve file in folder
+        for original_file_name in files:
+            if original_file_name.endswith(ORIGINAL_FILE_EXTENSION):
+                print(f"Found curve file {original_file_name}")
+                output_file_name = "{}{}".format(
+                    original_file_name.strip(ORIGINAL_FILE_EXTENSION), OUTPUT_FILE_EXTENSION
+                )
+                with open(os.path.join(root, original_file_name), mode='r', encoding='utf-8') as original_file, \
+                        open(os.path.join(output_folder, output_file_name), mode='w+', encoding='utf-8') as output_file:
+                    # Iterate over header block dictionary
+                    for header_title, header_block in isis_calibration.items():
+                        output_file.write(f'# {header_title}\n' + '# {\n')
+                        # Again iterate over the nested dictionary
+                        for entry, calib_value in header_block.items():
+                            output_file.write(f'#   "{entry}" : "{calib_value}"\n')
+                    # Strip first 6 lines from file
+                    for x in range(6):
+                        next(original_file)
+                    output_file.write("# }\n")
+                    # Go through rest of lines and add correct format to output file
+                    for original_line in original_file:
+                        original_line = original_line
+                        values = original_line.split()
+                        # Swap temp/resistance columns to match the calibration file format
+                        output_file.write(f"{values[1]},{values[0]}\n")
+
+                # We've found the dat file so don't need to look at any others in the directory
+                break
+        else:
+            print(f"Could not find {ORIGINAL_FILE_EXTENSION} file in {root}")
+
+    print(
+        "Finished converting, please check files and then copy and commit to the correct repo "
+        "https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Calibration-Files"
+    )
```

Comment:
Noticing that the majority of `convert_curve_files.py` is duplicated from `convert_temp_calib_files.py` and that there is a lot going on in `if __name__ == "__main__":`, could we instead rework this to make a script called `file_converter` or similar that follows more of an OOP methodology implementation of `convert_temp_calib_files.py` and `convert_curve_files.py` to break things up into classes and methods which can be called using either a `.curve` file or `.dat` file type?
Breaking the script up in this way would make it easier to add new file types going forward and also add unit tests either as tech-debt or otherwise.